sycl/include/sycl/buffer.hpp

```diff
@@ -168,8 +168,10 @@ template <typename T, int dimensions = 1,
               typename std::enable_if_t<(dimensions > 0) && (dimensions <= 3)>>
 class buffer : public detail::buffer_plain,
                public detail::OwnerLessBase<buffer<T, dimensions, AllocatorT>> {
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
```

Comment:
Why this change? If it's ultimately needed than we probably need an `#else` with the assert prior to the current one.

Reply:
The "old" sycl::vec implementation did not correctly support device_copyable for vec.  So this assert is only valid when using the "new" sycl::vec (ie, the -fpreview-breaking-changes flag).  There is no "else" as there was no assert before.

Reply:
https://github.com/intel/llvm/commit/68af512680ebd6de520fd54ab68847b23dedb209

Reply:
Yes, I know.  But that PR was done recently building atop the "new" sycl:;vec, that we are now pushing backwards.  The "old" sycl::vec was not device_copyable (and not conformant) and cannot sustain that assertion.

Reply:
nevermind - I'm being dumb as to what you are asking. Restored the assert we had previously.